calcom/cal.com
# fix: cal.ai OpenAI agent, add Groq option

## What does this PR do?

In Cal.ai, the prior #13953 looked like it was passing a prompt that would not have worked. I have updated it and additionally added an option to choose Groq-hosted models to power the agent.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Tests (Unit/Integration/E2E or any other test)
- This change requires a documentation update

## How should this be tested?

- Are there environment variables that should be set?
GROQ_API_KEY if using Groq

I was unable to figure out how to set up all the tools to properly test, and it additionally looked like there was no coverage for these endpoints. I was able to set up a rudimentary test Next.js endpoint under `apps/ai/src/app/api/test/route.ts`, which I basically ran as a script.

Here are some LangSmith traces for this script with OpenAI and Groq. The tool calls fail since I was unable to set them up properly, but the agent loop at least calls the tool, even if it later fails:

Trace for OpenAI: https://smith.langchain.com/public/c8d41632-d11d-45e9-9c8c-cf233fc4b876/r
Trace for Groq: https://smith.langchain.com/public/b8bf3cfc-4115-46e1-b8dd-fa9c13bb26f7/r

The proper way to do this would be to set up a test framework that runs CI and ideally online evals.

Groq does not support tool calling yet, so I have to use a less reliable and more generic agent for now.

## Mandatory Tasks
